low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,467 @@
+
+        # Integrate into the main dbus loop to catch events
+        DBusGMainLoop(set_as_default=True)
+
+        # Initialize SystemBus and get Manager Interface
+        self._bus = SystemBus()
+        systemd = self._bus.get_object("org.freedesktop.systemd1",
+                                       "/org/freedesktop/systemd1")
+        self._manager = Interface(systemd,
+                            dbus_interface='org.freedesktop.systemd1.Manager')
+
+        self.remove_disabled_services()
+
+        return True
+
+    def remove_disabled_services(self):
+        """Remove `disabled` services from the list of services to monitor."""
+        temp = self.services_to_monitor.copy()
+        for service in temp:
+            try:
+                if 'disabled' in str(self._manager.GetUnitFileState(service)):
+                    self.services_to_monitor.remove(service)
+            except DBusException as err:
+                logger.error(f"{service} removed from services_to_monitor due "\
+                             f"to error : {err}")
+                self.services_to_monitor.remove(service)
+
+    def run(self):
+        logger.info(f"Monitoring Services : {self.services_to_monitor}")
+        try:
+            # Register all the services to signal of 'PropertiesChanged' and
+            # raise an alert if some service is not active on initially or if
+            # Unit is not found for the service
+            services_to_monitor_copy = self.services_to_monitor.copy()
+            for service in services_to_monitor_copy:
+                err = self.connect_to_prop_changed_signal(service)
+                if err:
+                    self.raise_alert(service, "N/A", "N/A", "N/A", "N/A",
+                        "N/A", "N/A", 0)
+                    logger.error(f"{service} is not active initially. \n Error {err}")
+                else:
+                    self.services_to_monitor.remove(service)
+
+            logger.debug(f"failed_services : {self.failed_services}")
+            logger.debug(f"services_to_monitor : {self.services_to_monitor}")
+
+            # Retrieve the main loop which will be called in the run method
+            self._loop = gobject.MainLoop()
+
+            # Initialize the gobject threads and get its context
+            gobject.threads_init()
+            context = self._loop.get_context()
+
+            time_to_check_lists = self.current_time() + self.polling_frequency
+
+            while self.is_running():
+                # At interval of 'thread_sleep' check for events occured for
+                # registered services and process them(call on_pro_changed())
+                context.iteration(False)
+                time.sleep(self.thread_sleep)
+
+                # At interval of 'polling_freqency' process unregistered
+                # services and services with not-active state.
+                if time_to_check_lists <= self.current_time():
```

Comment:
this condition will never be false with logic on line 165

Reply:
Consider eg: current_time() = 0;
line 153:   time_to_check_lists = current_time (0)+ polling_frequency (15) = 15
line 163:   is time_to_check_lists (15) <= (0) current_time ??  false.
                till current_time becomes >= 15 this will be false;
when current_time is >=15   condition is  true.
line 164-165 :    time_to_check_lists = current_time (15) + polling_frequency (15) = 30
and continues...

Reply:
seems i misread, looks fine